postgresql/devpi_postgresql/main.py

```diff
@@ -183,17 +185,17 @@ def __init__(self, basedir, notify_on_commit, cache_size, settings=None):
             if key in settings:
                 setattr(self, key, settings[key])
 
-        if any(key in settings for key in ("ssl_cert_reqs", "ssl_ca_certs",
-                                           "ssl_certfile", "ssl_keyfile")):
-            self.ssl_opts = ssl_opts = {}
-            if "ssl_cert_reqs" in settings:
-                ssl_opts["cert_reqs"] = {
-                    "cert_optional": ssl.CERT_OPTIONAL,
-                    "cert_required": ssl.CERT_REQUIRED}.get(settings["ssl_cert_reqs"])
+        if any(key in settings for key in self.SSL_OPT_KEYS):
+            self.ssl_context = ssl_context = ssl.create_default_context(
+                cafile=settings.get('ssl_ca_certs'))
 
-            for key in ("ssl_ca_certs", "ssl_certfile", "ssl_keyfile"):
-                if key in settings:
-                    ssl_opts[key[4:]] = settings[key]
+            if 'ssl_certfile' in settings:
+                ssl_context.load_cert_chain(settings['ssl_certfile'],
+                                            keyfile=settings.get('ssl_keyfile'))
+
+            check_hostname = settings.get('ssl_check_hostname')
```

Comment:
This isn't in SSL_OPT_KEYS, is that an oversight?

Reply:
> Ok, looks like `ssl_cert_reqs` is gone and `ssl_check_hostname` is new, that change would certainly justify a major version bump.

Yes, I think I made a mistake to include `ssl_cert_reqs` in the first place because I didn't quite understand what it did. Based on my current reading of the standard library documentation, either value ("cert_optional" or "cert_required") would have had the same effect.

The addition of `ssl_check_hostname` is the primary purpose of this pull request. In contrast with my earlier pull request, secure connections will now default to checking host name. It may be desirable to disable this.

Reply:
> This isn't in SSL_OPT_KEYS, is that an oversight?

The ssl settings are now "ssl_check_hostname", "ssl_ca_certs", "ssl_certfile" and "ssl_keyfile" and they are all in SSL_OPT_KEYS. Did I miss something?